src/main.rs

```diff
@@ -1,30 +1,48 @@
-//! Provides the main entry point to the program.
+//! The main entry point for the `muse2` command-line tool.
 use ::log::info;
+use clap::{Arg, Command};
 use muse2::log;
 use muse2::model::Model;
 use muse2::settings::Settings;
-use std::env;
 use std::path::PathBuf;
 
-/// The main entry point to the program
+/// The main entry point for the `muse2 run` command.
 fn main() {
-    // Parse command-line arguments
-    let args: Vec<String> = env::args().collect();
-    if args.len() != 2 {
-        panic!("Must provide path to model folder.");
-    }
-    let model_dir = PathBuf::from(args[1].as_str());
+    let cmd = Command::new("muse2")
+        .version("2.0")
+        .about("MUSE2 Simulation Tool")
+        .subcommand(
+            Command::new("run").about("Run a model simulation").arg(
+                Arg::new("model_dir")
+                    .help("Path to the model directory")
+                    .required(true)
+                    .index(1)
+                    .value_parser(clap::value_parser!(PathBuf)),
+            ),
```

Comment:
As mentioned by @alexdewar , this should be its own function, so in the future we can easily do things like:
```suggestion
            define_command_A(),
            define_command_B(),
            define_command_C(),
```